components/CreateEventsWizard.tsx

```diff
@@ -0,0 +1,353 @@
+import "react-time-picker/dist/TimePicker.css";
+import "react-calendar/dist/Calendar.css";
+import DatePicker from "react-date-picker/dist/entry.nostyle";
+import TimePicker from "react-time-picker/dist/entry.nostyle";
+
+import { RRule } from "rrule";
+import { DateTime } from "luxon";
+import { CirclePicker } from "react-color";
+import ClipLoader from "react-spinners/ClipLoader";
+
+type CreateEventsWizardProps = {
+  handleClose: () => void;
+};
+
+const CreateEventsWizard: React.FC<CreateEventsWizardProps> = ({ handleClose }) => {
+  // create wizard states
+  const [name, setName] = useState("");
+  const [multipleLevels, setMultipleLevels] = useState(false);
+  const [minLevel, setMinLevel] = useState(1);
+  const [maxLevel, setMaxLevel] = useState(1);
+  const [startDate, setStartDate] = useState(new Date());
+  const [startTime, setStartTime] = useState("10:00");
+  const [endTime, setEndTime] = useState("11:00");
+  const [showColorPicker, setShowColorPicker] = useState(false);
+  const [showRepeatModal, setShowRepeatModal] = useState(false);
+  const [rruleText, setRruleText] = useState("Never Repeat");
+  const [color, setColor] = useState("#ffc702");
+  const [teachers, setTeachers] = useState("");
+  const [valid, setValid] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [errMsg, setErrMsg] = useState("");
+
+  // saved repeat modal states
+  const [repeat, setRepeat] = useState(false);
+  const [interval, setInterval] = useState(1);
+  const [weekDays, setWeekDays] = useState<number[]>([]);
+  const [endType, setEndType] = useState("never");
+  const [endDate, setEndDate] = useState(new Date());
+  const [count, setCount] = useState(1);
+
+  const client = useContext(APIContext);
+
+  const colorMap: { [name: string]: string } = {
+    "#ffc702": "yellow",
+    "#ef5da8": "magenta",
+    "#46d5b3": "mint",
+    "#5d5fef": "purple",
+    "#5ec0f0": "blue",
+    "#ff0202": "red",
+  };
+
+  // validates event wizard on field input
+  useEffect(() => {
+    setValid(true);
+    if (!name || !teachers) {
+      setValid(false);
+    }
+    if (!minLevel || (multipleLevels && (!maxLevel || minLevel >= maxLevel))) {
+      setValid(false);
+    }
+    if (!startDate || !startTime || !endTime) {
+      setValid(false);
+    }
+  }, [name, multipleLevels, minLevel, maxLevel, startDate, startTime, endTime, color, teachers]);
+
+  // force max level to exceed min level
+  useEffect(() => {
+    if (minLevel >= maxLevel) {
+      setMaxLevel(minLevel + 1);
+    }
+  }, [minLevel, maxLevel]);
+
+  // callback for hiding modal on close
+  const handleRepeatClose = (): void => {
+    setShowRepeatModal(false);
+  };
+
+  // callback for setting repeat modal states on save
+  const handleRepeatStates = (
+    repeat_param: boolean,
+    interval_param: number,
+    weekDays_param: number[],
+    endType_param: string,
+    endDate_param: Date,
+    count_param: number
+  ): void => {
+    setRepeat(repeat_param);
```

Comment:
again, the repeatParam is always going to be true, so this might require some refactoring. It should be a quick fix so long as we have default values for all the repeating parameters you need in this function